src/VerifyEmailHelper.php

```diff
@@ -63,21 +64,60 @@ public function generateSignature(string $routeName, string $userId, string $use
         return new VerifyEmailSignatureComponents(\DateTimeImmutable::createFromFormat('U', (string) $expiryTimestamp), $signature, $generatedAt);
     }
 
-    public function validateEmailConfirmation(string $signedUrl, string $userId, string $userEmail): void
+    public function validateEmailConfirmation(string $signedUrl, string $userId, string $userEmail, ?Request $request = null): void
```

Comment:
I think we'll be better off leaving this method alone and adding a deprecation `@trigger_deprecation('Use validateEmailConfirmationFromRequest instead...)`

1) theres alot of if/else action happening in here
2) we don't have a clean path forward to reorder the arguments where `Request` is the 1st argument.